### ISSUE TITLE
Get a node from the beginning of the UUID

### ISSUE THREAD

**Issue by giovannipizzi** [aiidateam/aiida_core#527]
I think it is useful to identify nodes just by the first characters of the UUID (the PK changes when exporting, and the full UUID is long). This is also what is commonly done by e.g. git.

A stub of the function to get the node could be:
```
def get_node(start_uuid):
    from aiida.orm.querybuilder import QueryBuilder

    q = QueryBuilder()
    q.append(Node, filters={'uuid': {'like': "{}%".format(start_uuid)}},
             project='*')
    # For efficiency, only retrieve up to two nodes
    q.limit(2)
    res = q.all()
    if len(res) == 0:
        raise ValueError("No node found with UUID starting by '{}'".format(
                start_uuid))
    if len(res) > 1:
        raise ValueError("More than one found with UUID starting by '{}'".format(
                start_uuid))
    return res[0][0]
```

I think this functionality could/should be directly into load_node(), by default, if a string is passed. Do you agree?

**Comment by giovannipizzi** [comment]
Note that the function above is not very good to deal with (optional) dashes in the UUID.
The following function deals much better with them.
Note that I first remove all dashes, and then leave an optional parameter to decide if we should add back the dashed at the right place. This is important because e.g. with django we store the UUID in the DB WITH dashes, so they need to be in place. On the other hand, probably the user would input the UUID without any dash.

Improved code:
```
def get_node(start_uuid, query_with_dashes=True):
    from aiida.orm.querybuilder import QueryBuilder

    # Put back dashes in the right place
    start_uuid = start_uuid.replace('-', '')

    if query_with_dashes:
        # Essential that this is ordered from largest to smallest!
        for dash_pos in [20,16,12,8]:
            if len(start_uuid) > dash_pos:
                start_uuid = "{}-{}".format(
                    start_uuid[:dash_pos], start_uuid[dash_pos:]
                    )

    q = QueryBuilder()
    q.append(Node, filters={'uuid': {'like': "{}%".format(start_uuid)}},
             project='*')
    # For efficiency, only retrieve up to two nodes
    q.limit(2)
    res = q.all()
    if len(res) == 0:
        raise ValueError("No node found with UUID starting by '{}'".format(
                start_uuid))
    if len(res) > 1:
        raise ValueError("More than one found with UUID starting by '{}'".format(
                start_uuid))
    return res[0][0]
```

**Comment by giovannipizzi** [comment]
Fixed in #571
